#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.33751892317254845 0.34078413060753343 -0.8774663257507441 1.717062867972366
  position -2.323358795307492 2.560859166646851 2.9723628943798697
  follow "Robot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -1.32 -1.36 -0.04
  floorSize 10 10
  floorAppearance Parquetry {
    type "light strip"
  }
}
Wall {
  translation -2.784 0.753 -0.04
  rotation 0 0 1 1.5708
  name "wall 1"
  size 1.18 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -2.508 1.608 -0.04
  rotation 0 0 1 0.7854053071795867
  name "wall 1(1)"
  size 0.8 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -1.89284 1.884 -0.04
  rotation 0 0 1 -3.1415853071795863
  name "wall 1(2)"
  size 0.7 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -1.418 1.358 -0.04
  rotation 0 0 1 -3.1415853071795863
  name "wall 1(3)"
  size 0.9 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -0.81 1.19 -0.04
  rotation 0 0 -1 -2.3562053071795863
  name "wall 1(4)"
  size 0.5 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -0.641 0.681 -0.04
  rotation 0 0 -1 -1.5708053071795867
  name "wall 1(5)"
  size 0.7 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -0.4725 0.171 -0.04
  rotation 0 0 -1 -2.3562053071795863
  name "wall 1(6)"
  size 0.5 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -0.0633003 0.000999919 -0.04
  rotation 0 0 -1 3.14159
  name "wall 1(7)"
  size 0.5 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.34663 0.169405 -0.04
  rotation 0 0 -1 -0.7854053071795866
  name "wall 1(8)"
  size 0.5 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.3085 0.71 -0.04
  rotation 0 0 -1 -0.7854053071795866
  name "wall 1(9)"
  size 0.7 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.918 0.95095 -0.04
  rotation 0 0 1 -3.1415853071795863
  name "wall 1(10)"
  size 0.76 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 1.31349 0.673 -0.04
  rotation 0 0 -1 -1.5708053071795867
  name "wall 1(11)"
  size 0.61 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.92352 -0.950682 -0.04
  rotation 0 0 -1 -1.5708053071795867
  name "wall 1(12)"
  size 0.25 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 1.26352 -0.660682 -0.04
  rotation 0 0 -1 -1.5708053071795867
  name "wall 1(13)"
  size 0.25 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.92352 -0.370682 -0.04
  rotation 0 0 -1 -1.5708053071795867
  name "wall 1(14)"
  size 0.25 0.05 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
CurvedRoadSegment {
  translation -3.384 -3.55 -0.0399
  rotation 0 0 1 -1.5707953071795862
  name "road 1"
  width 0.2
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  curvatureRadius 1
  totalAngle 3.14
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
CurvedRoadSegment {
  translation -1.384 -3.55 -0.0399
  rotation 0 0 1 1.5708
  name "road 1(1)"
  width 0.2
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  curvatureRadius 1
  totalAngle 1.57
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
CurvedRoadSegment {
  translation 0.096 -1.55 -0.0399
  rotation 0 0 1 -1.5707953071795862
  name "road 1(2)"
  width 0.2
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  curvatureRadius 1
  totalAngle 1.57
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
CurvedRoadSegment {
  translation 0.094 -1.543 -0.039
  rotation 0 0 1 -1.5707953071795862
  name "road 1(3)"
  width 0.04
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  curvatureRadius 1
  totalAngle 1.57
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -2.86 -0.04 -0.0399
  width 0.4
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0.8 0 0
  ]
  splineSubdivision 0
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -2.46002 -0.640332 -0.0399
  rotation 0 0 1 -1.571595307179586
  name "road(2)"
  width 0.2
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -2.42554 -1.56103 -0.0399
  rotation 0 0 1 -2.356995307179586
  name "road(3)"
  width 0.2
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0.8 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -2.92284 -2.104 -0.0399
  rotation 0 0 1 3.14079
  name "road(4)"
  width 0.2
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -3.84764 -2.06214 -0.0399
  rotation 0 0 1 -2.356995307179586
  name "road(5)"
  width 0.2
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0.8 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -4.38443 -2.55679 -0.0399
  rotation 0 0 -1 1.5716
  name "road(6)"
  width 0.2
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation 0.106815 -2.552 -0.0399
  rotation 0 0 1 3.14079
  name "road(7)"
  width 0.2
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    1.5 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation 1.02773 -1.44733 -0.0399
  rotation 0 0 1 3.14079
  name "road(8)"
  width 0.2
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -0.03315 -2.49649 -0.0399
  rotation 0 0 -1 -1.5699953071795862
  name "road(9)"
  width 0.2
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    1.15 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation 1.0976 -1.55746 -0.0399
  rotation 0 0 -1 -1.5699953071795862
  name "road(10)"
  width 0.2
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    2 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    baseColor 0 0 0
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -2.46002 -0.640332 -0.039
  rotation 0 0 1 -1.571595307179586
  name "road(11)"
  width 0.04
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0.96 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -2.45373 -1.59156 -0.039
  rotation 0 0 1 -2.356995307179586
  name "road(12)"
  width 0.04
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0.73 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -3.91201 -2.106 -0.039
  rotation 0 0 -1 0.000798985
  name "road(13)"
  width 0.04
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0.028 0 0
    0.96 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -3.87573 -2.09269 -0.039
  rotation 0 0 1 -2.356995307179586
  name "road(14)"
  width 0.04
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0.72 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -4.38165 -2.58873 -0.039
  rotation 0 0 1 -1.571595307179586
  name "road(15)"
  width 0.04
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0.96 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -1.39154 -2.54113 -0.039
  rotation 0 0 -1 0.000798985
  name "road(16)"
  width 0.04
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    1.53 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -0.030644 -1.42232 -0.039
  rotation 0 0 1 -1.571595307179586
  name "road(17)"
  width 0.04
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    1.23 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -0.04484 -1.44231 -0.039
  rotation 0 0 -1 0.000798985
  name "road(18)"
  width 0.04
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    1.241 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation 1.09379 -1.551 -0.039
  rotation 0 0 1 1.57
  name "road(19)"
  width 0.04
  numberOfLanes 1
  lines [
    RoadLine {
      type "continuous"
      width 0.03
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    1.69 0 0
  ]
  splineSubdivision 1
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Road {
  translation -2.86 -0.44 -0.0399
  name "road(1)"
  width 0.4
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0.8 0 0
  ]
  appearance DEF black_road PBRAppearance {
    metalness 0
  }
  pavementAppearance Pavement {
    type "black stone"
  }
  bottomTexture [
    ""
  ]
  locked FALSE
  roadBoundingObject TRUE
}
Robot {
  translation -2.46168 -0.576888 0.00969231
  rotation -0.00015330098766248564 -0.002852619770423936 0.9999959195213012 1.57
  children [
    LED {
      translation 0 -0.04 0.02
      children [
        DEF led_shape Shape {
          appearance PBRAppearance {
          }
          geometry Capsule {
            height 0.015
            radius 0.012
          }
        }
      ]
      name "led_left"
      boundingObject USE led_shape
      physics DEF led_physics Physics {
      }
      locked TRUE
    }
    LED {
      translation 0 0.003 0.02
      children [
        DEF led_shape Shape {
          appearance PBRAppearance {
          }
          geometry Capsule {
            height 0.015
            radius 0.012
          }
        }
      ]
      name "led_mid"
      boundingObject USE led_shape
      physics DEF led_physics Physics {
      }
      locked TRUE
      color [
        1 1 0
      ]
    }
    LED {
      translation 0 0.045 0.02
      children [
        DEF led_shape Shape {
          appearance PBRAppearance {
          }
          geometry Capsule {
            height 0.015
            radius 0.012
          }
        }
      ]
      name "led_right"
      boundingObject USE led_shape
      physics DEF led_physics Physics {
      }
      locked TRUE
      color [
        0 1 0
      ]
    }
    DistanceSensor {
      translation -0.15 0.06 -0.02
      rotation 0 1 0 1.57
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      type "infra-red"
    }
    DistanceSensor {
      translation -0.15 -0.01 -0.02
      rotation 0 1 0 1.57
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_mid_left"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      type "infra-red"
    }
    DistanceSensor {
      translation -0.15 0.01 -0.02
      rotation 0 1 0 1.57
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_mid_right"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      type "infra-red"
    }
    DistanceSensor {
      translation -0.15 -0.06 -0.02
      rotation 0 1 0 1.57
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      type "infra-red"
    }
    DistanceSensor {
      translation -0.15 -0.07 0
      rotation -0.7071067531399365 0.7071067531399365 -0.0002816519016716981 3.14103
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left_wall"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      lookupTable [
        0 0 0
        0.4 1000 0
      ]
    }
    DistanceSensor {
      translation -0.15 0.07 0
      rotation 0.7071067532046255 0.7071067532046255 0.0002813269018108966 3.14103
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right_wall"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      lookupTable [
        0 0 0
        0.3 1000 0
      ]
    }
    DistanceSensor {
      translation -0.15 0 0
      rotation 0 1 0 3.1408
      children [
        DEF ds_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.33 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_front_wall"
      boundingObject USE ds_shape
      physics DEF ds_physics Physics {
      }
      locked TRUE
      lookupTable [
        0 0 0
        0.3 1000 0
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 78.80792277698615
        axis 0 -1 0
        anchor -0.07 -0.09 0.01
      }
      device [
        RotationalMotor {
          name "front_left_motor"
        }
      ]
      endPoint Solid {
        translation -0.07 -0.09 0.01
        rotation 0.09489371660307924 -0.7041961156110326 -0.7036355685350109 3.330963320567136
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        name "solid(3)"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 104.56909707410935
        axis 0 -1 0
        anchor -0.07 0.09 0.01
      }
      device [
        RotationalMotor {
          name "front_right_motor"
        }
      ]
      endPoint Solid {
        translation -0.07 0.09 0.01
        rotation -0.3445199691856697 -0.6637737192471767 0.6638602567326013 3.805081392423087
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 106.77611436676486
        axis 0 -1 0
        anchor 0.07 0.09 0.01
      }
      device [
        RotationalMotor {
          name "rear_right_motor"
        }
      ]
      endPoint Solid {
        translation 0.07 0.09 0.01
        rotation -0.3443972452822498 -0.6641130973974055 0.6635844568005056 3.8054327691803804
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 82.85851496384298
        axis 0 -1 0
        anchor 0.07 -0.09 0.01
      }
      device [
        RotationalMotor {
          name "rear_left_motor"
        }
      ]
      endPoint Solid {
        translation 0.07 -0.09 0.01
        rotation 0.13314033042209034 -0.7010905302319279 -0.7005324552325973 3.4065248784310795
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 1 0.33 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
            }
          }
        ]
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    DEF robo_shape Shape {
      appearance PBRAppearance {
        baseColor 0 1 1
        metalness 0
      }
      geometry Box {
        size 0.3 0.14 0.04
      }
    }
  ]
  name "Robot"
  boundingObject USE robo_shape
  physics DEF robo_physics Physics {
  }
  controller "RoboController"
}
